apps/data/datalayer/src/email.rs

```diff
@@ -1,23 +1,60 @@
+use std::env;
+
 use axum::{http::StatusCode, routing::post, Json, Router};
+use lettre::{AsyncSmtpTransport, AsyncTransport, Message, Tokio1Executor};
 use serde::{Deserialize, Serialize};
+use tower_http::catch_panic::CatchPanicLayer;
 
-#[allow(dead_code)]
 #[derive(Deserialize, Debug, Serialize)]
 struct Email {
     to: String,
     subject: String,
     body: String,
 }
 
+pub fn email_endpoint() -> String {
+    env::var("EMAIL_ENDPOINT").expect("EMAIL_ENDPOINT env var not set")
+}
+
 pub fn email_router() -> Router {
-    Router::new().route("/api/email", post(email_handler))
+    Router::new()
+        .route("/api/email", post(email_handler))
+        .layer(CatchPanicLayer::new())
 }
 
 async fn email_handler(Json(email): Json<Email>) -> StatusCode {
     println!("Sending Email: {:?}", email);
+
+    if env::var("SEND_EMAILS").unwrap_or_default() == "TRUE" {
```

Comment:
Does this default to empty string? Doesn't matter since it's guarded, but curious.

Reply:
Yep. I originally had `unwrap_or(String::new())` but the linter wanted this instead.